doc/cprover-manual/jbmc-user-manual.md

```diff
@@ -0,0 +1,237 @@
+\ingroup module_hidden
+\page jbmc-user-manual JBMC User Manual
+
+
+\section jbmc-manual-intro Introduction
+
+JBMC is a bounded model checking tool for the Java language. It follows the
+same principles as CBMC, thus the existing
+[documentation for CBMC](http://www.cprover.org/cprover-manual/cbmc.html)
+largely applies to JBMC.
+
+In brief, JBMC creates a model from Java Bytecode (given as input as a .class
+file or a .jar) and performs static analysis and symbolic simulation which allow
+to prove or refute properties. These properties are either automatically
+generated, or provided by the user in the form of assertions.
+
+Using examples, the following section will guide you through the basic
+features of JBMC, and explore the most useful features.
+
+
+\section jbmc-tutorial JBMC tutorial
+
+\subsection jbmc-manual-auto-assert Automatic assertions
+
+Consider the following simplistic `ExampleArray.java` file which we have compiled
+into `ExampleArray.class` and which contains a utility method to query a given
+array at a given index:
+```
+ 1| package tutorial;
+ 2|
+ 3| public class ExampleArray {
+ 4|
+ 5|     public static int query(int[] array, int index) {
+ 6|         return array[index];
+ 7|     }
+ 8| }
+```
+
+Now let's run the following command to let JBMC tell us about potential errors
+in our `query` method.
+
+    $ jbmc ExampleArray.class --function tutorial.ExampleArray.query
```

Comment:
tutorial/ExampleArray.class

Reply:
As mentioned in a previous comment, there is no `tutorial` folder. We're not in a Maven project. If you just use `javac` on a java file (which is the approach chosen in this tutorial), you get a class file in the same directory.

Reply:
But this is not the way `java` works. You cannot run `java ExampleArray`, you must use `java tutorial.ExampleArray`. JBMC will work the same way soon. We can apply a simple 's#jbmc tutorial/ExampleArray.class#jbmc tutorial.ExampleArray' to this tutorial after this change.

Reply:
OK, got it. That makes sense now. I now consider the class file to be in a folder named after the package.